pkg/apiserver/webhooks.go

```diff
@@ -47,29 +48,76 @@ import (
 )
 
 const (
-	webhookServicePort = 8443
-	serverEndpoint     = "/validate-deletion"
+	serverEndpoint = "/validate-deletion"
 )
 
-func setupWebhooks(ctx context.Context, webhookNs string, certStorageDir string) error {
-	caBytes, err := createCerts(webhookNs, certStorageDir)
+type WebhookType string
+
+const (
+	WebhookTypeService WebhookType = "service"
+	WebhookTypeUrl     WebhookType = "url"
+)
+
+// WebhookConfig defines the configuration for the PackageRevision deletion webhook
+type WebhookConfig struct {
+	Type             WebhookType
+	ServiceName      string // only used if Type == WebhookTypeService
+	ServiceNamespace string // only used if Type == WebhookTypeService
+	Host             string // only used if Type == WebhookTypeUrl
+	Path             string
+	Port             int32
+	CertStorageDir   string
+}
+
+func NewWebhookConfig() *WebhookConfig {
+	var cfg WebhookConfig
+	// NOTE: CERT_NAMESPACE is supported for backward compatibility.
+	// TODO: We may consider using only WEBHOOK_SERVICE_NAMESPACE instead.
```

Comment:
Could you write an issue for removing it?
Also is this the right way of issuing certs? I mean tls in not mandatory, the webhooks can be created with insecrueSkipTlsVerify as well for dev envs. 
For production envs, I think using an external cert management entity would be better (see #53 ). For dev envs, could it be just disabled?

Reply:
I thought about this, and decided to leave this question to SIG Security :) and out-of-scope of this PR. the goal of this PR is to make the end-to-end tests work in the developement environment, and I think that is achieved.

to be honest, since you have to have the proper access to create a ValidationWebhookConfiguration object, it seems fine to me if the CA bundle in it generated on-the-fly, since it is only used for the communication between the kube-apiserver and porch. but again, I am not a security expert so I will let the experts decide :) and #53 will iron out the security aspect